src/instance.rs

```diff
@@ -25,18 +25,130 @@ pub trait PyObjectWithGIL: Sized {
 #[doc(hidden)]
 pub trait PyNativeType: PyObjectWithGIL {}
 
+/// A reference of type `T` which is in Python heap.
+///
+/// We can't implement `ToPyPointer` or `ToPyObject` for `pyclass`es, because they're not python
+/// objects untill copied to Python heap. So, instead of treating `&pyclass` as a Python object, we
+/// need to use special reference types `PyRef` and `PyRefMut`.
+/// ```
```

Comment:
This should be separated by an `# Example` header like in the std docs.

Reply:
:+1: